include/deal.II/base/tensor_base.h

```diff
@@ -854,28 +836,59 @@ void Tensor<0,dim,Number>::serialize(Archive &ar, const unsigned int)
   ar &value;
 }
 
-/*---------------------------- Inline functions: Tensor<1,dim,Number> ------------------------*/
 
+/*-------------------- Inline functions: Tensor<rank,dim> --------------------*/
 
-template <int dim, typename Number>
+
+namespace internal
+{
+  // TODO: Think about refactoring this into the TableIndices class as a
+  // general, polymorphic for extracting an item out of an object with
+  // nested identifiers.
+  template<int rank_> struct TensorIndicesHelper
+  {
+    // used for implementing Tensor<rank,dim>::operator[] with TableIndices
+    // tail recursive call to form up access to
+    //   tensor[indices[0]][indices[1]]...[indices[rank_]]
+    template<int rank, int dim, typename Number>
+    static inline
+    Number &extract(Tensor<rank_,dim,Number> &t, const TableIndices<rank> &indices)
```

Comment:
Well, that's called *overloading* with functions :-)

I think you should be able to do the same with just an overload set. But it's an internal interface, so I don't particularly care whether it is done or not. I really only care about the public interfaces ;-)